solang-parser/src/pt.rs

```diff
@@ -653,28 +658,52 @@ pub enum AssemblyStatement {
     Leave(Loc),
     Break(Loc),
     Continue(Loc),
+    Block(Loc, Box<AssemblyStatement>),
+    FunctionDefinition(Box<AssemblyFunctionDefinition>),
+    FunctionCall(Box<AssemblyFunctionCall>),
 }
 
 #[derive(Debug, PartialEq, Clone)]
 pub enum AssemblyExpression {
-    BoolLiteral(Loc, bool),
-    NumberLiteral(Loc, BigInt),
-    HexNumberLiteral(Loc, String),
-    StringLiteral(StringLiteral),
+    BoolLiteral(Loc, bool, Option<Identifier>),
+    NumberLiteral(Loc, BigInt, Option<Identifier>),
+    HexNumberLiteral(Loc, String, Option<Identifier>),
+    StringLiteral(StringLiteral, Option<Identifier>),
     Variable(Identifier),
     Assign(Loc, Box<AssemblyExpression>, Box<AssemblyExpression>),
     LetAssign(Loc, Box<AssemblyExpression>, Box<AssemblyExpression>),
-    Function(Loc, Box<AssemblyExpression>, Vec<AssemblyExpression>),
+    FunctionCall(Box<AssemblyFunctionCall>),
     Member(Loc, Box<AssemblyExpression>, Identifier),
     Subscript(Loc, Box<AssemblyExpression>, Box<AssemblyExpression>),
 }
 
-pub type AssemblyBlock = Vec<AssemblyStatement>;
+#[derive(Debug, PartialEq, Clone)]
+pub struct AssemblyTypedIdentifier {
+    pub loc: Loc,
+    pub identifier_name: Identifier,
```

Comment:
When naming a field in a struct, repeating the name of the struct in fields in unnecessary. These fields can be called `name` and  `type`.

Reply:
`type` is a reserved word in Rust. I can't name the field as `type`, so I am changing to `assembly_type`.

Reply:
This issue comes up in other parts of the parse tree. In other parts we have used `ty` instead.